### PR TITLE
fix: text input overflow on focus

### LINKED ISSUE #5909: 🐛 Bug:  UI/UX Tidy - Form layout
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

- [ ] Long text field when used in form has alignment & overflow issues
Screenshot 2023-06-16 at 5 34 44 PM

- [ ] Disable dark mode until we fix it right; currently icons are barely visible in form view

### Desired Behavior

x

### Project Details

Node: **v16.14.0**
Arch: **arm64**
Platform: **darwin**
Docker: **false**
RootDB: **sqlite3**
PackageVersion: **0.109.1**

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by sreehari2003**
## Change Summary

- closes #5909 

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Before
Screenshot 2023-08-02 at 10 53 29 PM

## After

Screenshot 2023-08-02 at 10 41 12 PM

### ONFOCUS
Screenshot 2023-08-02 at 10 40 47 PM

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
Failed in Shared Form View.

image

**Comment by sreehari2003** [comment]
> Failed in Shared Form View.
> 
> image

fixed issue , took me more time to find the file than fixing the issue

**Review by wingkwong — CHANGES_REQUESTED** [review]
Please also fix the padding in Shared Form View.

image

image


image

File: packages/nc-gui/components/smartsheet/Form.vue [hunk 1]
```diff
@@ -730,7 +730,7 @@ watch(view, (nextView) => {
                     <LazySmartsheetDivDataCell class="relative">
                       <LazySmartsheetCell
                         v-model="formState[element.title]"
-                        class="nc-input"
+                        class="nc-input truncate"
                         :class="`nc-form-input-${element.title.replaceAll(' ', '')}`"
```
Comment (sreehari2003) [inline packages/nc-gui/components/smartsheet/Form.vue:734]: i tried fixing that but looks like the css `nc-form-input-longtext` is over riding my css , i tried making my css !important but not working , what should i do ? i tried searching for property `nc-form-input-longtext` but couldnt find anywhere

**Comment by sreehari2003** [comment]
> > Failed in Shared Form View.
> > image
> 
> fixed issue

## I have also tried to complete #5969   here  , right now entering tab wont trigger form submission

**Comment by wingkwong** [comment]
@sreehari2003 

1. Don't misuse header, i.e. `##` 
2. Issue 5969 has been assigned to someone else. You should ask for an assignment request if he/she is idle before taking over the task to avoid duplicate work.
3. Please don't submit other changes that are not related to the original issue, i.e. issue 5909. Supposedly you should create different branches for different issues, not everything in develop branch. Each PR should only target to one issue.

File: packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue [hunk 2]
```diff
@@ -165,7 +165,7 @@ const onDecode = async (scannedCodeValue: string) => {
                       <LazySmartsheetCell
                         v-else
                         v-model="formState[field.title]"
-                        class="nc-input"
+                        class="nc-input truncate"
                         :data-testid="`nc-form-input-cell-${field.label || field.title}`"
                         :class="`nc-form-input-${field.title?.replaceAll(' ', '')}`"
```
[the remark this answers is not in the document]
Comment (wingkwong) [reply packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue:184]: The CSS overriding yours is here. I think this block can be removed.

https://github.com/nocodb/nocodb/blob/8694bf2f49979e3b7132bb751b2358692cd1571e/packages/nc-gui/pages/%5BprojectType%5D/form/%5BviewId%5D/index.vue#L123-L125

File: packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue [hunk 3]
```diff
@@ -165,10 +179,11 @@ const onDecode = async (scannedCodeValue: string) => {
...
                         :column="field"
+                        :style="{ paddingLeft: '0.5rem !important' }"
```
Comment (wingkwong) [inline packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue:186]: See the above comment. This line is not a correct fix.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (sreehari2003) [reply packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue:170]: ok

File: packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue (hunk 3 above)
Comment (sreehari2003) [reply packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue]: > See the above comment. This line is not a correct fix.

done , that CSS overriding wouldn't be happening if the style tag was scoped right ?

Comment (wingkwong) [reply packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue]: FYI, `!` means `!important`. `!p-0` overrides every padding. Didn't see the reason we need that so removing that block should resolve the padding issue.

Comment (sreehari2003) [reply packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue]: ok got it , this is ready for merge right ?

**Review by wingkwong — APPROVED** [review]

**Comment by PyJim** [comment]
I thought the issue was supposed to be reserved for interns at Code Day labs?

**Comment by PyJim** [comment]
I thought the issue was supposed to be reserved for interns at Code Day labs?

**Comment by wingkwong** [comment]
@PyJim I don't see there is an assignee marked on that issue. Besides, there's no update on your PR for over a week and the changes are not what we want. The issue does affect the UX and it's been open for 3 months. It shouldn't take too much time.

**Pull request merged**